Automattic/woocommerce-payments
# Jetpack Tracking integration

Fixes #3823

#### Changes proposed in this Pull Request

This PR integrates `jetpack-tracking` package to support the tracking needs of platform checkout.

`Tracking` class that comes from `jetpack-tracking` package is extended to 
* Not track IP address and User-Agent
* Track guest users

**See** pdpOw2-l8-p2, pdpOw2-mz-p2

Proposed changes only implement the infrastructure needed to track C2 events. Actual tracking events will be implemented in subsequent PRs. Event names mentioned in the testing instructions below are for testing purposes only. They might not fire from the place mentioned below and will likely track other properties. 







## Testing instructions





* In your `wp_options` table ensure that
    * `_wcpay_feature_platform_checkout` option is set to 1.
    * `jetpack_tos_agreed` option is set to 1. This will already be set if you have Jetpack plugin installed
* Enable Platform Checkout option as a merchant from Payments -> Settings -> Express checkouts

### Test tracking from PHP

#### Setup
* Add the following hook to the constructor `add_action( 'woocommerce_add_to_cart', [ $this, 'track_add_to_cart' ], 10, 6 );`


#### As a logged in normal user
* Login to the store as a customer (or as any user except administrator)
* Add a product to the cart
* In ~5 minutes, checks the Tracks Live View for an event called, `woocommerceanalytics_order_checkout_start`
* Expand the event Props section
* Notice the `userId` 

#### As a logged out user
Repeat the above tests and notice that `woocommerceanalytics_order_checkout_start` appears on live view with a different `userId`

#### As site administrator
* Login to the store as site admin 
* Add a product to the cart
* Ensure that no tracking event is created on MC live view after ~5 minutes

#### Disable Platform Checkout
* Disable Platform Checkout option as a merchant from Payments -> Settings -> Express checkouts
* Add a product to the cart as a normal user
* Ensure that no tracking event is created on MC live view after ~5 minutes

### Test tracking from JS
* Add the following tracking line within the `onClick` handler in https://github.com/Automattic/woocommerce-payments/blob/develop/client/components/platform-checkout/index.tsx#L47
```
wcpayTracks.recordUserEvent( 'platform_checkout_registered', null );
``` 
* Make sure wcpayTracks is imported with `import wcpayTracks from 'tracks';` as well
* Add a product to the cart 
* Go to blocks checkout page
* Click "Buy now with platform checkout". 
* In ~5 minutes, checks the Tracks Live View for an event called, `woocommerceanalytics_platform_checkout_registered`
 

-------------------


**Post merge**





cc @brianyu28
